src/main/java/edu/ucdavis/dss/ipa/api/components/course/CourseViewController.java

```diff
@@ -538,6 +544,27 @@ public CourseView createMultipleCoursesFromDW(@RequestBody List<SectionGroupImpo
 
 						activityService.saveActivity(activity);
 					}
+
+					if (importAssignments) {
+						for (DwInstructor dwInstructor : dwSection.getInstructors()) {
+							DwPerson dwPerson = dwRepository.getPersonByLoginId(dwInstructor.getLoginId());
+
+							if (dwPerson == null) {
+								log.warn("getPersonByLoginId Response from DW returned null, for criterion = " + dwInstructor.getLoginId());
+								continue;
+							}
+
+							String instructorEmail = dwPerson.getEmail();
+
+							// Find or create an instructor
+							Instructor instructor = instructorService.findOrCreate(dwInstructor.getFirstName(), dwInstructor.getLastName(), instructorEmail, dwInstructor.getLoginId(), workgroupId);
+							instructor = instructorService.save(instructor);
+
+							// Find or create a teachingAssignment
+							TeachingAssignment teachingAssignment = teachingAssignmentService.findOrCreateOneBySectionGroupAndInstructor(sectionGroup, instructor);
+							teachingAssignment.setApproved(true);
```

Comment:
teachingAssignment needs to be saved after setting approved to true, otherwise the change won't be persisted in the database